src/core/filters/filtermetadata.js

```diff
@@ -20,6 +20,19 @@ export default class FilterMetadata {
      * @type {string}
      */
     this.displayValue = displayValue;
+
+    /**
+     * The origin component type of this filter, such as FilterOptions or
+     * DateRangeFilter.
+     * @type {string}
+     */
+    this.originComponent = originComponent;
+
+    /**
+     * The type of filter this is, either 'RADIUS_FILTER' or 'STATIC_FILTER'.
+     * @type {string}
+     */
+    this.optionType = optionType || 'STATIC_FILTER';
```

Comment:
Neither optionType nor originComponent is expressive enough on it's own. It seems like what we really want is some higher-level construct, "filterType". This type can be: "Facet", "Distance", "Combined", "NLP", "Geolocation", "Range", etc. When computing the analytics event for removing a filter, we can maintain a mapping of filterType to component type. I think having such a granular "filterType" could come in handy. I noticed the spec said one example of "removed_from_component" was "VerticalResults". When would we see that?

Reply:
hmm I missed that, no idea what they mean

Reply:
VerticalResults definitely doesn't make any filters

Reply:
We discussed last night no longer needing originComponent. I've asked Max a question about optionType. I'm not sure what the value of that attribute is supposed to be for Facets. Is it unset, is it STATIC_FILTER? I'm not sure.

I think filterType would still be a useful thing to add because it gives us more information than just optionType. optionType just allows us to differentiate between two kinds of static filters. But, we can have dynamic filters, nlp filters, etc. There are also more than just two kinds of static filter. filterType would allow us to encapsulate all that in one value (instead of originComponent + optionType). 

We could also add neither filterType nor optionType right now. By looking at the structure of the filter, you can differentiate between normal static and distance filter options, right? A distance filter is just a number, correct? Other filters are objects. I'd be fine if we didn't add either optionType or filterType for now, with the plan being adding filterType later on.

Reply:
added filtertype which is either STATIC, FACET, NLP, or RADIUS!